modules/core/04-channel/keeper/upgrade.go

```diff
@@ -782,15 +782,15 @@ func (k Keeper) startFlushing(ctx sdk.Context, portID, channelID string, upgrade
 	}
 
 	upgrade.LatestSequenceSend = nextSequenceSend - 1
-	upgrade.Timeout = getUpgradeTimeout()
+	upgrade.Timeout = k.getUpgradeTimeout(ctx)
 	k.SetUpgrade(ctx, portID, channelID, *upgrade)
 
 	return nil
 }
 
-// TODO: use a hard coded value for now. Will be resolved in https://github.com/cosmos/ibc-go/issues/4313
-func getUpgradeTimeout() types.Timeout {
-	return types.NewTimeout(clienttypes.NewHeight(1, 1000), 0)
+// getUpgradeTimeout returns the timeout for the given upgrade.
+func (k Keeper) getUpgradeTimeout(ctx sdk.Context) types.Timeout {
+	return k.GetParams(ctx).UpgradeTimeout
```

Comment:
nice, I think we maybe want to create a new timeout which is offset by the values in the timeout stored in params. 

The block height and timestamp values in the timeout stored in params are absolute values - maybe we should be adding the timeout height to the current block height from `ctx` as well as adding the timeout timestamp to the current block time from `ctx`.

what do you think?